### PR TITLE
♻️ Complete architecture refactor: .NET 10, MVVM, DI, tests

### PR DESCRIPTION
**Pull request by sasler**
## Complete Architecture Refactor

This PR refactors the entire CafeMaestro codebase from a vibe-coded prototype into a properly architected .NET MAUI application.

### Changes across 6 phases (15 commits):

**Phase 0: Platform Upgrade**
- Upgraded from .NET 9 to .NET 10 MAUI
- Added CommunityToolkit.Mvvm 8.4.2 + CommunityToolkit.Maui 14.0.1
- Created xUnit test project with MAUI-compatible configuration

**Phase 1: Service Layer**
- Extracted 7 service interfaces (IAppDataService, IBeanDataService, IRoastDataService, etc.)
- Replaced manual ServiceProvider resolution with constructor injection across all pages
- Replaced reflection-based event manipulation with IDisposable SuspendNotifications pattern

**Phase 2: Model Improvements**
- Added validation (Validate/IsValid) to BeanData, RoastData, RoastLevelData with TDD
- Moved RoastLevelViewModel to ViewModels/ folder, converted to CommunityToolkit.Mvvm ObservableObject

**Phase 3: Shared Services**
- Extracted CsvParserService consolidating ~370 lines of duplicated CSV parsing
- Created NavigationService with centralized route constants
- Added AlertService for ViewModel-driven dialog interactions

**Phase 4: MVVM Conversion**
- Created ViewModels for all 9 pages using CommunityToolkit.Mvvm source generators
- Extracted ~4,000 lines from code-behind into proper ViewModels
- RoastPageViewModel alone handles timer, bean selection, first crack, weight tracking, save/edit

**Phase 5-6: Polish**
- Replaced hardcoded colors with theme resource references
- Code cleanup, stale comment removal
- Fixed 2 bugs found by code review (timer event leak, locale-dependent parsing)

### Stats
- **75 files changed**, +7,830 / -5,652 lines
- **87 unit tests** (was 0)
- ✅ Windows + Android builds pass with 0 errors

### CODE REVIEW

File: CafeMaestro/Models/BeanData.cs [hunk 1]
```diff
@@ -32,12 +34,52 @@ public class BeanData
...
         public bool IsOutOfStock => RemainingQuantity <= 0;
 
+        [JsonIgnore]
+        public bool IsValid => !Validate().Any();
+
         public BeanData()
         {
             // Set remaining quantity to match initial quantity by default
             RemainingQuantity = Quantity;
         }
```
[the remark this answers is not in the document]
Comment (sasler) [reply CafeMaestro/Models/BeanData.cs:42]: Good catch. Removed the misleading comment — the constructor assignment `RemainingQuantity = Quantity` was always setting 0=0 since Quantity defaults to 0 before object initializer runs. The constructor body is now empty. Fixed in e8c6653.

File: CafeMaestro/Models/BeanData.cs [hunk 2]
```diff
@@ -32,12 +34,52 @@ public class BeanData
...
+            var errors = new List<string>();
+
+            if (Country is null)
+            {
+                errors.Add("Country must not be null.");
+            }
+
+            if (CoffeeName is null)
+            {
+                errors.Add("CoffeeName must not be null.");
```
[the remark this answers is not in the document]
Comment (sasler) [reply CafeMaestro/Models/BeanData.cs]: Valid bug — since Country and CoffeeName default to empty strings, the \is null\ checks would never trigger. Changed both to \string.IsNullOrWhiteSpace()\ and updated error messages accordingly. Also added a dedicated test for the empty/whitespace validation scenario. Fixed in e8c6653.

File: CafeMaestro/CafeMaestro.csproj [hunk 3]
```diff
@@ -22,6 +22,7 @@
         <SingleProject>true</SingleProject>
         <ImplicitUsings>enable</ImplicitUsings>
         <Nullable>enable</Nullable>
+        <LangVersion>preview</LangVersion>
```
[the remark this answers is not in the document]
Comment (sasler) [reply CafeMaestro/CafeMaestro.csproj]: Agreed — changed to \latest\ in both the main project and test project. Verified Windows and Android builds pass with 0 errors. The CommunityToolkit.Mvvm source generators work fine without preview features on .NET 10. Fixed in e8c6653.
